src/Utility/ExprFunctions.C

```diff
@@ -87,7 +92,56 @@ static void ExprException (const ExprEval::Exception& exc, const char* task,
 }
 
 
-int EvalFunc::numError = 0;
+/*!
+  \brief Static helper converting an index pair into a single index.
+  \details Assuming Voigt notation ordering; 11, 22, 33, 12, 23, 13.
+*/
+
+int voigtIdx (int d1, int d2)
+{
+  if (d1 > d2)
+    std::swap(d1,d2); // Assuming symmetry
+
+  if (d1 < 1 || d2 > 3)
+    return -1; // Out-of-range
+  else if (d2-d1 == 0) // diagonal term, 11, 22 and 33
+    return d1-1;
+  else if (d2-d1 == 1) // off-diagonal term, 12 and 23
+    return d2+1;
+  else // off-diagonal term, 13
+    return 5;
+}
+
+
+/*!
+  \brief Static helper that splits a function expression into components.
+*/
+
+std::vector<std::string> splitComps (const std::string& functions,
+                                     const std::string& variables)
+{
+  std::vector<std::string> comps;
+  size_t pos1 = functions.find("|");
+  size_t pos2 = 0;
+  while (pos2 < functions.size())
+  {
+    std::string func(variables);
+    if (!func.empty() && func[func.size()-1] != ';')
+      func += ';';
+    if (pos1 == std::string::npos)
+      func += functions.substr(pos2);
+    else
+      func += functions.substr(pos2,pos1-pos2);
+    comps.push_back(func);
+    pos2 = pos1 > 0 && pos1 < std::string::npos ? pos1+1 : pos1;
+    pos1 = functions.find("|",pos1+1);
+  }
+
+  return comps;
+}
+
+
```

Comment:
Maybe enough with one blank line between the two `}` lines here

Reply:
ok.